includes/Modules/Idea_Hub.php

```diff
@@ -439,10 +441,55 @@ protected function create_data_request( Data_Request $data ) {
 			case 'GET:saved-ideas':
 				return $this->fetch_ideas( 'saved' );
 			case 'POST:update-idea-state':
-				// @TODO implementation
-				return function() {
-					return null;
-				};
+				if ( ! isset( $data['name'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'name' ),
+						array( 'status' => 400 )
+					);
+				}
+
+				if ( ! isset( $data['saved'] ) && ! isset( $data['dismissed'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						__( 'Either "saved" or "dismissed" parameter must be provided.', 'google-site-kit' ),
+						array( 'status' => 400 )
+					);
+				}
+
+				$idea_name       = $data['name'];
+				$idea_name_parts = explode( '/', $data['name'] );
+
+				$parent = $this->get_parent_slug();
+				$parent = sprintf(
+					'%s/ideaStates/%s',
+					untrailingslashit( $parent ),
+					array_pop( $idea_name_parts )
+				);
+
+				$update_mask = array();
+
+				$body = new Google_Service_Ideahub_GoogleSearchIdeahubV1alphaIdeaState();
+				$body->setName( $idea_name );
+
+				if ( isset( $data['saved'] ) ) {
+					$body->setSaved( filter_var( $data['saved'], FILTER_VALIDATE_BOOLEAN ) );
+					$body->setDismissed( false );
```

Comment:
Now that we've figured out the use of the update mask, is this even still relevant? I think it can now be removed.

Reply:
Yes, we don't need it. Added it to #3818